rust/src/tests/lib/aggregator.rs

```diff
@@ -0,0 +1,60 @@
+use crate::{
+    aggregator::service::{Aggregator, ServiceHandle as InnerServiceHandle, ServiceRequests},
+    common::client::Credentials,
+};
+use bytes::Bytes;
+use std::{future::Future, pin::Pin};
+
+#[derive(Clone)]
+pub struct ServiceHandle(InnerServiceHandle);
+
+pub struct ByteAggregator {
+    weights: Vec<u8>,
+}
+
+impl ByteAggregator {
+    pub fn new() -> ByteAggregator {
+        ByteAggregator { weights: vec![] }
+    }
+}
+
+impl Aggregator for ByteAggregator {
+    type Error = ();
+
+    type AddWeightsFut = Pin<Box<dyn Future<Output = Result<(), ()>> + Send>>;
+    type AggregateFut = Pin<Box<dyn Future<Output = Result<Bytes, ()>> + Send>>;
```

Comment:
I don't think we need a `Pin<Box<_>>` here. Since the computation is actually synchronous a `futures::future::Ready` should work.